src/xtgeo/grid3d/_gridprop_import_eclrun.py

```diff
@@ -568,3 +565,35 @@ def import_gridprops_from_restart_file(
     if close:
         filehandle.close()
     return read_properties
+
+
+def import_gridprop_from_init(pfile, name, grid, fracture=False):
+    init_props = import_gridprop_from_init_file(pfile.file, [name], grid, fracture)
+    if len(init_props) != 1:
+        raise ValueError(f"Could not find property {name} in {pfile}")
+    init_props[0]["name"] = decorate_name(
+        init_props[0]["name"], grid.dualporo, fracture
+    )
+    return init_props[0]
+
+
+def import_gridprop_from_restart(
```

Comment:
I think this function could benefit from type-hints, docstring and perhaps a test.

Do I read this right that `date` can be of some kind of format like "2012-26-07", and "20122607" and one of "all", "first", "last"?
If so, why?

Reply:
Yes, date can be any one of those, and the reason is to avoid breaking changes.

Reply:
Added typehint.

Reply:
Also, "all" is not a valid option, so removed that.

Reply:
"all" still seems to be an option for both `find_gridprops_from_restart_file` and `find_gridprops_from_restart_file_sections` since typehints `Literal["all"]` are used.
Are you thinking about something else?

By the way, `Literal[]` is new to me and it's cool so thanks :)

Reply:
Yes, "all" is still an option for "find". I think the confusion is that the "find" functions are reused to also find more than one gridproperty.